src/xiaomi_ble/parser.py

```diff
@@ -1000,9 +1000,25 @@ def obj3003(
     return result
 
 
-# The following data objects are device specific. For now only added for
-# LYWSD02MMC, MJWSD05MMC, XMWSDJ04MMC, XMWXKG01YL, LINPTECH MS1BB(MI), HS1BB(MI),
-# K9BB, PTX https://miot-spec.org/miot-spec-v2/instances?status=all
+# The following data objects are device specific.
+# https://miot-spec.org/miot-spec-v2/instances?status=all
+def obj4801(
+    xobj: bytes, device: XiaomiBluetoothDeviceData, device_type: str
+) -> dict[str, Any]:
+    """Temperature"""
+    (temp,) = struct.unpack("f", xobj)
```

Comment:
All of these could be written to avoid constructing the struct each time.

F_UPACK = struct.Struct("f").unpack

other examples https://github.com/Bluetooth-Devices/inkbird-ble/blob/1afa92eade4e13a6c5913a4e761b35b433400c06/src/inkbird_ble/parser.py#L56